pkg/agent/util/net_unix.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux darwin
+
+package util
+
+import (
+	"fmt"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"k8s.io/klog"
+	"net"
+)
+
+// GetLocalNodeAddr return a local IP/mask that is on the path of default route.
+func GetDefaultLocalNodeAddr() (*net.IPNet, netlink.Link, error) {
+	conn, err := net.Dial("udp", "8.8.8.8:80")
+	if err != nil {
+		return nil, nil, err
+	}
+	defer conn.Close()
+	localIP := conn.LocalAddr().(*net.UDPAddr).IP
```

Comment:
this code does not require internet access. UDP is connectionless.  As network connect udp socket, in queries route table in how to reach internet(8.8.8.8), but doing so it finds out what is the local ip and local IP used for that udp socket. and there is no packet sent out or expected reply back.

Of course the code will be problematic is default route and route to 8.8.8.8 used different outbound IP and dev ...

Hope this helps, Su

Reply:
let us say there is a default rt, and default has dev associated it, and there must be a single IP  on that device, otherwise why do u know which one is default?

This code is not invented by me, please see thread for various discussion.  https://stackoverflow.com/questions/23558425/how-do-i-get-the-local-ip-address-in-go.

To me this is most precise solution of the bunch.

Reply:
U are correct that if there are multiple interfaces on the node, the IP picked by default route may not be best to use to communicate with some peer nodes;  That will perhaps be another discussion how to handle multiple IPs on the worker node? 

regarding to node agent having route to internet. Yes, the underlying assumption is that worker node should have a default route. can u think of scenario where worker node does not require default route?

Reply:
we now uses k8s ip to retrieve interface IP/mask